### DISCUSSION TITLE
launch_core1 does not work

### DISCUSSION THREAD

**Discussion by Ing-Dom** [earlephilhower/arduino-pico#314] [Q&A]
Hi,

I wanted to manually lauch core1 with the SDK function `multicore_launch_core1(loop_core1);`
I do this at the end of setup();

`void loop_core1()
{
    //_hwsensors.Loop();
    delay(200);
    digitalWrite(10, HIGH);
    delay(200);
    digitalWrite(10, LOW);
}`

It does not work. It seems that loop_core1 is not executed.

when I rename loop_core1 to loop1 and do not lauch ot manually, it works.
Why does the manual launching not work? Any idea?

The reason I want to do this is that I want to setup everything first and then start the 2. core. If they are started simultaenusly i would have to add some sync.

edit: sorry, I just got it... I forgot the while(true)...

**Comment by RISC-E-Biscuit** [comment]
From the arduino pico library examples. Seems to work.

// Demonstrates a simple use of the setup1()/loop1() functions
// for a multiprocessor run.

// Will output something like, where C0 is running on core 0 and
// C1 is on core 1, in parallel.

// 11:23:07.507 -> C0: Blue leader standing by...
// 11:23:07.507 -> C1: Red leader standing by...
// 11:23:07.507 -> C1: Stay on target...
// 11:23:08.008 -> C1: Stay on target...
// 11:23:08.505 -> C0: Blue leader standing by...
// 11:23:08.505 -> C1: Stay on target...
// 11:23:09.007 -> C1: Stay on target...
// 11:23:09.511 -> C0: Blue leader standing by...
// 11:23:09.511 -> C1: Stay on target...
// 11:23:10.015 -> C1: Stay on target...

// Released to the public domain

// The normal, core0 setup
void setup() {
  Serial.begin(115200);
  delay(5000);
}

void loop() {
  Serial.printf("C0: Blue leader standing by...\n");
  delay(1000);
}

// Running on core1
void setup1() {
  delay(5000);
  Serial.printf("C1: Red leader standing by...\n");
}

void loop1() {
  Serial.printf("C1: Stay on target...\n");
  delay(500);
}
